### ISSUE TITLE
Backlog not updated, change specific paper ID

### ISSUE THREAD

**Issue by ankitT20** [ankushgarg1998/ipu-ranklist#164]
Request to Please read the issue patiently it's confusing but very simple solution, also apologies for unprofessional language.

For results of batch 2021-2025 of USAR, EDC (University School of Automation & robotics)
Students affected more than 111 over 3 branches (AI-DS, IIOT, AR)

Issues:
1. wrong 3rd semester -  Credits Obtained:   27 / 30, wrong SGPA
2. wrong Overall Result - Credits Obtained:   140 / 143, wrong CGPA, wrong Percentage, etc.

Explanation:
Around 150 students gave reappear exam for ABS201 subject of 3 credits (LINEAR AND ABSTRACT ALGEBRA) Paper ID 516201, 
after the reappear result particularly for subject ABS201, the original Paper ID is same for all, but after reappear result they were changed.
AIML - original 516201, reappear 516201
AIDS - original 516201, reappear 519201
IIOT  - original 516201, reappear 517201
AR    - original 516201, reappear 520201
So for AIML its correct, but for others issues start.
also I noted that these new paper IDs were of junior batch 2022-2026.

Profile Links of all branches for reappear and passed of ABS201 subject-
AIML (correct)- https://www.ipuranklist.com/student/00419011621
AIDS (issues) - https://www.ipuranklist.com/student/02219011921
IIOT  (issues) - https://www.ipuranklist.com/student/01119011721
AR   (issues)  - https://www.ipuranklist.com/student/00119012021

Solution 1: manually change paper IDs for 3 ipu results pdf
In other words **change paper IDs 517201,519201,520201 back to its original 516201** for only 3 pdf only where ABS201 subject code is mentioned, not for BS201 (that is for junior batch with different subject name)
can be done for specific SchemeID (just one value), This way it can be guaranteed to change only the affected students, and not disturb any other (juniors) and future results.

GGSIPU results links for reappear of 3rd sem of all branches-
AIML (NO CHANGE needed) : http://ggsipu.ac.in/ExamResults/2024/210224/RTS_202312_516_S01_S03_S05_21022024.pdf 
AIDS SchemeID: 195192021001 Page 6: http://ggsipu.ac.in/ExamResults/2024/210224/RTS_202312_519_S01_S03_S05_21022024.pdf 
IIOT SchemeID: 195172021001 Page 6: http://ggsipu.ac.in/ExamResults/2024/210224/RTS_202312_517_S01_S03_S05_21022024.pdf 
AR SchemeID: 195202021001 Page 6: http://ggsipu.ac.in/ExamResults/2024/210224/RTS_202312_520_S01_S03_S05_21022024.pdf 

SchemeID: 195192021001 Page 6 is same for all branches, change paper IDs 517201,519201,520201 back to its original 516201.

_ignore if solution 1 accepted: also the batches are already shown differently like AIDS for 2021-2025, AIDS2 for 2022-2026, similarly AIML AIML2, IIOT IIOT2, AR AR2 can be seen on opening dropdown menu of branch in College Ranklist navbar_
https://www.ipuranklist.com/ranklist/usit-btech?batch=21&sem=3&college=USAR&branch=IOT&shift=0

**conclusion: just change a single value (paper IDs 517201,519201,520201 back to its original 516201) on page 6 for three results.**
Total modifications needed: 3

**Comment by ankushgarg1998** [comment]
Hi @ankitT20 

Thank you for raising this. It's commendable how well you've researched and have provided complete details of the issue.

Please note that your subject should never have had paper_id: 516201.
It was released as 517201, when your first 3rd sem result was announced. It is a bug in the current implementation of IPU Ranklist that picks up a different possible combination of a paper_id for a paper_code i.e. ABS201 in your case, and many other cases.

This messes up some of the reappearing students' results.
I'm already working on a fix for this, which was being tracking in #150 . Due to it being a massive change in the DB schema itself, it's taking quite some time. It should be live before the end of this quarter which will fix yours and all other such results.

I'll keep this issue open and provide an update once this is resolved.

**Comment by ankitT20** [comment]
Absolutely, sorry for missing that detail, I did not checked the original result from 07-Jun-2023

AIML 516201
AIDS 519201
IIOT  517201
AR    520201

Confirmed, the Paper ID was never 516201 for IIOT and the **reappear results are correct**, 
**the original were wrong**.

Thanks. you are a genius man.


_just to include links of original results
AIML: http://ggsipu.ac.in/ExamResults/2023/080623/RTG_202301_516_S03_00_01.pdf
AIDS: http://ggsipu.ac.in/ExamResults/2023/080623/RTG_202301_519_S03_00_01.pdf
IIOT: http://ggsipu.ac.in/ExamResults/2023/080623/RTG_202301_517_S03_00_01.pdf
AR: http://ggsipu.ac.in/ExamResults/2023/080623/RTG_202301_520_S03_00_01.pdf

So basically PaperID = Prg. Code: 517 + Paper Code
and the same Prg. Code in SchemeID also: 19 517 2021 001 nice_

**Comment by ankushgarg1998** [comment]
Hi @ankitT20 

This fix for this issue has been deployed. ✅ 
Please verify. Thanks.
